wire-ios/WireCommonComponents/Analytics/WireAnalytics.swift

```diff
@@ -16,5 +16,39 @@
 // along with this program. If not, see http://www.gnu.org/licenses/.
 //
 
+import Foundation
+import WireSystem
+
 /// Namespace for analytics tools.
-public enum WireAnalytics {}
+public enum WireAnalytics {
+
+    private static let isSetUpLock = NSLock()
+    private static var isSetUp = false
+
+    public static func setup() {
+        // Adding a lock here since some app extension might execute this setup in the same process as the main app.
+        // https://stackoverflow.com/a/62674277
+        isSetUpLock.lock()
+        defer { isSetUpLock.unlock() }
+
+        guard !isSetUp else {
+            assertionFailure("WireAnalytics.setup() called more than once")
```

Comment:
I guess if https://stackoverflow.com/a/62674277 happens we might not want to assert here... It is normal behavior.

Reply:
True, let's change it if it happens during debugging.
For release builds this statement will have no effect.